File: front/src/modelo/compra.ts
import Cliente from "./cliente"
import Pet from "./pet"
import Produto from "./produto"
import Servico from "./servico"

export default interface Compra {
    id?: number
    cliente: Cliente
    pet: Pet
    tipo: string
    compradoPro?: Produto
    compradoSer?: Servico
}

File: back/pl-atv-5/src/main/java/com/fatec/pl/modelo/Compra.java
package com.fatec.pl.modelo;

import org.springframework.hateoas.RepresentationModel;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
public class Compra extends RepresentationModel<Compra> {
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;
	
	@Column
	private Cliente cliente;
	
	@Column
	private Pet pet;
	
	@Column
	private String tipo;
	
	@Column
	private Produto compradoPro;
	
	@Column
	private Servico compradoSer;
}
